#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.4775337037872902 -0.8427069568728518 -0.24860922465847912 1.1067868379180985
  position -1.687599802565442 2.8588042673679146 2.562287398490236
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0.1 0
  rotation -1 0 0 -5.307179586466759e-06
  children [
    DEF jointFR HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.12 0 0.12
      }
      device [
        PositionSensor {
          name "position_FR"
        }
        RotationalMotor {
          name "motor_FR"
          maxVelocity 100
        }
      ]
      endPoint DEF FR Solid {
        translation 0.12 0 0.12
        rotation -0.9999999999999999 0 0 1.5707953071795862
        children [
          DEF FR Shape {
            appearance Appearance {
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid_RL"
        boundingObject USE FR
        physics Physics {
        }
      }
    }
    DEF jointFL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.12 0 -0.12
      }
      device [
        PositionSensor {
          name "position_FL"
        }
        RotationalMotor {
          name "motor_FL"
          maxVelocity 100
        }
      ]
      endPoint DEF FL Solid {
        translation 0.12 0 -0.12
        rotation -0.9999999999999999 0 0 1.5707953071795862
        children [
          DEF FL Shape {
            appearance Appearance {
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid_FL"
        boundingObject USE FL
        physics Physics {
        }
      }
    }
    DEF jointBL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.12 0 -0.12
      }
      device [
        PositionSensor {
          name "position_BL"
        }
        RotationalMotor {
          name "motor_BL"
          maxVelocity 100
        }
      ]
      endPoint DEF BL Solid {
        translation -0.12 0 -0.12
        rotation -0.9999999999999999 0 0 1.5707953071795862
        children [
          DEF BL Shape {
            appearance Appearance {
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid_BL"
        boundingObject USE BL
        physics Physics {
        }
      }
    }
    DEF jointFL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.12 0 0.12
      }
      device [
        PositionSensor {
          name "position_BR"
        }
        RotationalMotor {
          name "motor_BR"
          maxVelocity 100
        }
      ]
      endPoint DEF BR Solid {
        translation -0.12 0 0.12
        rotation -0.9999999999999999 0 0 1.5707953071795862
        children [
          DEF BR Shape {
            appearance Appearance {
            }
            geometry Cylinder {
              height 0.03
              radius 0.05
            }
          }
        ]
        name "solid_BR"
        boundingObject USE BR
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.0784314 0.741176 0.8
        }
      }
      geometry DEF bodybox Box {
        size 0.3 0.05 0.2
      }
    }
  ]
  boundingObject USE bodybox
  physics Physics {
  }
  controller "car_mpc_controller"
  supervisor TRUE
}
Solid {
  translation 0 0.07 0.46
  children [
    DEF box Shape {
      appearance Appearance {
        material Material {
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  boundingObject USE box
  physics Physics {
  }
}
